### PR TITLE
Feature/issue 363

### LINKED ISSUE #363: Adjust change password flow
see https://github.com/gnosis/safe/tree/master/specs/change_password

### PR DESCRIPTION
**Pull request by elgatovital**
Closes #363: Adjust change password flow

Changes proposed in this pull request:
- Use activity & fragments instead of a dialog
- Use custom view for bottom panel

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/pm/gnosis/heimdall/ui/settings/general/changepassword/ChangePasswordViewModel.kt [hunk 1]
```diff
@@ -12,10 +14,23 @@ import javax.inject.Inject
 class ChangePasswordViewModel @Inject constructor(
     private val encryptionManager: EncryptionManager
 ) : ChangePasswordContract() {
-    override fun validatePassword(password: String): Single<Result<Collection<PasswordValidationCondition>>> =
+
+    private val confirmSubject = PublishSubject.create<Unit>()
+    private val viewState = PublishSubject.create<ViewState>()
+    private var _viewState = ViewState(State.ENTER_OLD_PASSWORD, true)
```
Comment (rmeissner) [inline app/src/main/java/pm/gnosis/heimdall/ui/settings/general/changepassword/ChangePasswordViewModel.kt]: We never read `_viewState`, do we need it?

File: app/src/main/java/pm/gnosis/heimdall/ui/settings/general/changepassword/ChangePasswordEnterNewFragment.kt [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    @Inject
+    lateinit var viewModel: ChangePasswordContract
+
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        var root = inflater.inflate(R.layout.layout_password_change_enter_new, container, false)
+        return root
+    }
+
+    @SuppressLint("RestrictedApi")
```
Comment (rmeissner) [inline app/src/main/java/pm/gnosis/heimdall/ui/settings/general/changepassword/ChangePasswordEnterNewFragment.kt]: Which restricted api do we use here?

File: app/src/main/java/pm/gnosis/heimdall/views/TwoStepPanel.kt [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+import androidx.core.widget.ImageViewCompat
+import com.jakewharton.rxbinding2.view.clicks
+import io.reactivex.Observable
+import kotlinx.android.synthetic.main.layout_two_step_panel.view.*
+import pm.gnosis.heimdall.R
+import pm.gnosis.heimdall.utils.setColorFilterCompat
+import pm.gnosis.svalinn.common.utils.getColorCompat
+
+
+class TwoStepPanel @JvmOverloads constructor(
```
Comment (rmeissner) [inline app/src/main/java/pm/gnosis/heimdall/views/TwoStepPanel.kt]: Just out of curiosity, why choose a custom layout vs just 2 image views?

**Comment by rmeissner** [comment]
what about the deprecated code? can it be removed?

File: app/src/main/java/pm/gnosis/heimdall/ui/settings/general/changepassword/ChangePasswordViewModel.kt (hunk 1 above)
Comment (elgatovital) [reply app/src/main/java/pm/gnosis/heimdall/ui/settings/general/changepassword/ChangePasswordViewModel.kt]: no, removing

File: app/src/main/java/pm/gnosis/heimdall/views/TwoStepPanel.kt (hunk 3 above)
Comment (elgatovital) [reply app/src/main/java/pm/gnosis/heimdall/views/TwoStepPanel.kt]: colouring logic was too cluttery. This way everything is hidden in one place. It makes other layouts that use this panel little bit more short & readable.

File: app/src/main/java/pm/gnosis/heimdall/ui/settings/general/changepassword/ChangePasswordEnterNewFragment.kt (hunk 2 above)
Comment (elgatovital) [reply app/src/main/java/pm/gnosis/heimdall/ui/settings/general/changepassword/ChangePasswordEnterNewFragment.kt]: none, removing

**Pull request merged**